samples/matter/lock/Kconfig.sysbuild

```
@@ -10,20 +10,13 @@ config NRF_DEFAULT_MULTIPROTOCOL
 config NRF_DEFAULT_BLUETOOTH
 	default y if BOARD_NRF7002DK_NRF5340_CPUAPP
 
+config MATTER_FACTORY_DATA
+	default y if !BOARD_NRF21540DK
+
 choice BOOTLOADER
-	default BOOTLOADER_MCUBOOT
+	default BOOTLOADER_MCUBOOT if !BOARD_NRF21540DK
 endchoice
 
-if BOARD_NRF21540DK
-
-config BOOTLOADER_MCUBOOT
-	default n
-
-config BOOTLOADER_NONE
-	default y
-
-endif
-
 if BOOTLOADER_MCUBOOT && (BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_NRF7002DK_NRF5340_CPUAPP)
 
 config MCUBOOT_UPDATEABLE_IMAGES
```

```suggestion
config MCUBOOT_UPDATEABLE_IMAGES
	default 3 if NRF_WIFI_PATCHES_EXT_FLASH_STORE
```
Then the `sample.matter.lock.thread_wifi_switched.smp_dfu` target can be built.

---

Issue was in sysbuild Kconfig, has been fixed